clang_format.py

```diff
@@ -115,29 +161,64 @@ def clang_format_path() -> Path:
     return clang_format_file
 
 
-def main() -> None:
+def get_version_list(
+    versionmap: Mapping[
+        Tuple[int, int, int],
+        Mapping[str, str],
+    ]
+) -> Tuple[str, ...]:
+    data = [tuple(map(str, tup)) for tup in versionmap.keys()]
+    return tuple([".".join(version) for version in data])
```

Comment:
You could represent this as a one-liner:

```python
choices=tuple(".".join(f"{i}" for i in k) for k in versionmap.keys()),
```

Reply:
Changes implemented as per above comments.